src/bid/encoding.rs

```diff
@@ -0,0 +1,243 @@
+//! Encoding module for Bid structure.
+//!
+//! See: https://hackmd.io/@7dpNYqjKQGeYC7wMlPxHtQ/BkfS78Y9L
+
+use super::errors::BidError;
+use super::Bid;
+use dusk_bls12_381::Scalar;
+use dusk_plonk::constraint_system::{StandardComposer, Variable};
+use failure::Error;
+use poseidon252::{sponge::sponge::*, StorageScalar};
+
+const BYTE_PADDING_LEN: usize = 28;
+
+/// Encodes a `Bid` in a `StorageScalar` form by applying the correct padding
+/// and encoding methods stated in https://hackmd.io/@7dpNYqjKQGeYC7wMlPxHtQ/BkfS78Y9L
+/// and collapsing it into a `StorageScalar` which can be then stored inside of a
+/// `kelvin` tree data structure.
+pub(crate) fn tree_leaf_encoding(bid: &Bid) -> Result<StorageScalar, Error> {
```

Comment:
I thought about it. The problem is: Where will this trait be defined on? 

Since what makes more sense is to have it in kelvin, in the PoseidonTree structure

Reply:
There is no PoseidonTree in kelvin, kelvin is just the backend, it only knows of general annotations and hashes.

But this will come up in the refactor as well, let's see